frontend/src/metabase/admin/settings/components/widgets/PersistingModelsToggleWidget.tsx

```diff
@@ -0,0 +1,47 @@
+import React from "react";
+import { t } from "ttag";
+
+import Toggle from "metabase/core/components/Toggle";
+
+import { PersistedModelsApi } from "metabase/services";
+
+type Props = {
+  setting: {
+    key: string;
+    value: boolean | string;
+    default: number;
+    options: Record<string, string>;
+  };
+  disabled?: boolean;
+  onChangeSetting: (setting: string, value: unknown) => void;
+};
+
+const PersistingModelsToggleWidget = ({
+  setting,
+  onChangeSetting,
+  disabled,
+}: Props) => {
+  const value = setting.value == null ? setting.default : setting.value;
+  const on = value === true || value === "true";
+
+  async function onChange() {
+    if (disabled) {
+      return;
+    }
+    if (on) {
+      await PersistedModelsApi.disablePersistence();
+    } else {
+      await PersistedModelsApi.enablePersistence();
+    }
+    onChangeSetting(setting.key, !on);
+  }
+
+  return (
+    <div className="flex align-center pt1">
```

Comment:
Maybe use `boolean` field type instead of custom widget? 
<img width="659" alt="Screenshot 2022-04-08 at 22 13 09" src="https://user-images.githubusercontent.com/8542534/162509148-2c22a11b-530b-4171-88c4-308b5135a297.png">